backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java

```diff
@@ -284,15 +291,21 @@ public AddTestResultResponse addMultiplexResult(
     TestOrder order =
         _testOrderRepo.fetchQueueItem(org, person).orElseThrow(TestOrderService::noSuchOrderFound);
 
-    DeviceSpecimenType deviceSpecimenType =
-        _deviceTypeService.getDeviceSpecimenType(deviceSpecimenTypeId);
+    DeviceType deviceType = _deviceTypeService.getDeviceType(deviceTypeId);
+    SpecimenType specimenType =
+        deviceType.getSwabTypes().stream()
+            .filter(swab -> swab.getInternalId().equals(specimenTypeId))
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new IllegalGraphqlArgumentException(
+                        "invalid device type and specimen type combination"));
 
     lockOrder(order.getInternalId());
-
     TestOrder savedOrder = null;
 
     try {
-      order.setDeviceSpecimen(deviceSpecimenType);
+      order.setDeviceTypeAndSpecimenType(deviceType, specimenType);
```

Comment:
re my previous comment: we set the device and specimen types for the order here, but then the constructor called on line 316 below ultimately calls the `TestEvent` constructor that will now use the facility defaults

Reply:
think this causes a bug

https://user-images.githubusercontent.com/6401323/213825450-3b41693d-0469-4d7e-b5e3-769e70feba90.mov

Reply:
great catch! thank you!

Reply:
addressed in this [commit](https://github.com/CDCgov/prime-simplereport/pull/5012/commits/a8011f68ef8d765100a3ad325e5f13efae19700f)